templates/CRM/Core/BillingBlock.tpl

```diff
@@ -28,7 +28,7 @@
   {if $paymentFields|@count}
     <fieldset class="billing_mode-group {$paymentTypeName}_info-group">
       <legend>
-        {$paymentTypeLabel}
+        {ts}{$paymentTypeLabel}{/ts}
```

Comment:
I think it should be written like this @mlutfy can you confirm?

```suggestion
        {ts 1=$paymentTypeLabel}{%1}{/ts}
```

Reply:
No, neither is correct. Please find the source of the string, and translate it there. This may happen to work, but it's the wrong way to use gettext, and would throw a warning.

From a quick grep:

```
$ grep -r paymentTypeLabel
templates/CRM/Core/BillingBlock.tpl:        {$paymentTypeLabel}
CRM/Core/Payment/Form.php:    $form->assign('paymentTypeLabel', self::getPaymentLabel($processor['object']));
CRM/Core/Payment/Form.php:    $paymentTypeLabel = self::getPaymentTypeLabel($processor);
CRM/Core/Payment/Form.php:    return $isVisible ? $paymentTypeLabel : '';
```

which eventually leads to: https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/Payment/Form.php#L209

```
    return ts('%1 Information', [$paymentProcessor->getPaymentTypeLabel()]);
```

And this is also bad, because it makes assumptions about how the string will be concatenated :(

Reply:
to clarify: `$paymentProcessor->getPaymentTypeLabel()` can return `Credit Card`, or it can return `Unicorn`.

and then `ts('Unicorn Information')`  won't resolve to something that is translatable, not to mention that `%1 Information` will already have been translated, so we happen not to see the bug in some languages (ex: French), but it will still be a problem for other languages.

cc @eileenmcnaughton